package/kernel/mac80211/files/lib/wifi/mac80211.sh

```diff
@@ -1,6 +1,16 @@
 #!/bin/sh
 append DRIVERS "mac80211"
 
+if json_is_a system object; then
+	json_select system
+		local wifi_ssid wifi_key wifi_country
+		json_get_vars wifi_ssid wifi_key wifi_country
+	json_select ..
+fi
+
+secure_ssid=$([ -n "$wifi_ssid" ] && echo $wifi_ssid || ( uci -q get system.@system[-1].label_macaddr || cat /sys/class/net/eth0/address) | tr [UPPER] [LOWER] | tr ':' '_')
+secure_psk=$( [ -n "$wifi_key"  ] && echo $wifi_key  || ( uci -q get system.@system[-1].label_macaddr || cat /sys/class/net/eth0/address) | tr [UPPER] [LOWER] | tr ':' '_')
```

Comment:
Please use more than one line here to enhance readability.

If you want to use label MAC address, rely on the function get_mac_label in /lib/functions/system.sh
(only a fraction of addresses is stored in uci).